src/clients/HubPoolClient.ts

```diff
@@ -264,13 +264,15 @@ export class HubPoolClient extends BaseAbstractClient {
   l2TokenHasPoolRebalanceRoute(l2Token: Address, l2ChainId: number, hubPoolBlock = this.latestHeightSearched): boolean {
     return Object.values(this.l1TokensToDestinationTokensWithBlock).some((destinationTokenMapping) => {
       return Object.entries(destinationTokenMapping).some(([_l2ChainId, setPoolRebalanceRouteEvents]) => {
-        return setPoolRebalanceRouteEvents.some((e) => {
-          return (
-            e.blockNumber <= hubPoolBlock &&
-            e.l2Token.truncateToBytes20() === l2Token.truncateToBytes20() &&
-            Number(_l2ChainId) === l2ChainId
-          );
-        });
+        // sort events descending by block number
+        const sortedEvents = sortEventsDescending(setPoolRebalanceRouteEvents).filter(
```

Comment:
Minor q. on performance - the majority of these events are thrown away straight after they're sorted. Filtering for l2ChainId first would for instance cut the number of events down significantly.